### PR TITLE
Issue 5509: Make styles of details summaries and fieldset legends consistent

### LINKED ISSUE #5509: Be consistent with the style of details summaries and fieldset legends
While working on #5503, I noticed that fieldset legends have `text-decoration` on hover, while the summary link on `<details>` has no underline on hover.

We should be consistent with this, so either remove the underline from fieldsets, or add it to details.

* `<fieldset>` element: See "Administration" (or any other project group) on the page at `admin/modules/list` uses http://backdrop.local/core/themes/seven/css/style.css (approx line 723?)
* `<details>` element: See "How layouts work" at the top of the page at `admin/structure/layouts` uses http://backdrop.local/core/modules/system/css/system.theme.css (approx line 15)

Proposed solution:
* remove the underline from fieldset `<legend> <a>`  (on hover) to more closely match `<details>` styling.

### PR DESCRIPTION
**Pull request by RakeshPotnuru**
Fixes https://github.com/backdrop/backdrop-issues/issues/5509

No underline on hover:
Screenshot 2022-03-25 142046

Same arrows:
image
image

### CODE REVIEW

File: core/modules/layout/css/layout.admin.css [hunk 1]
```diff
@@ -395,3 +395,25 @@
...
+
+details summary {
+  list-style-type: none;
+}
+details[open] summary span::before, details summary span::before {
+  content: "";
+  position: relative;
+  left: -.3em;
+  top: -.2em;
+  width: 0;
```
Comment (stpaultim) [inline core/modules/layout/css/layout.admin.css:407]: Are these two items necessary? I experimented with them and did not see that they are having any effect.

File: core/modules/layout/css/layout.admin.css [hunk 2]
```diff
@@ -395,3 +395,25 @@
...
+  height: 0;
```
Comment (stpaultim) [inline core/modules/layout/css/layout.admin.css:408]: Are these two items necessary? I experimented with them and did not see that they are having any effect.
```suggestion
  width: 0;
  height: 0;
```

Comment (stpaultim) [inline core/modules/layout/css/layout.admin.css]:
```
  width: 0;
  height: 0;
```
@RakeshPotnuru  Are these two items necessary? I tried to disable them in the inspector and did not see any change.

Comment (RakeshPotnuru) [reply core/modules/layout/css/layout.admin.css]: I thought they will help override any default browser styles. But not, will remove.

**Pull request closed without merging**